Firestore/Swift/Source/Codable/third_party/FirestoreDecoder.swift

```diff
@@ -330,15 +333,22 @@ private struct _FirestoreKeyedDecodingContainer<K: CodingKey>: KeyedDecodingCont
   }
 
   public func decode<T: Decodable>(_: T.Type, forKey key: Key) throws -> T {
-    if T.self == AutoPopulatedDocumentId.self {
+    if T.self == SelfDocumentID.self {
+      let docRef = decoder.userInfo[
+        Firestore.Decoder.documentRefUserInfoKey!
+      ] as! DocumentReference?
+
       if contains(key) {
-        throw FirestoreDecodingError.fieldNameConfict("Field name conflict:" +
-          " \(key.stringValue) is an `AutoPopulatedDocumentId` but " +
-          "also exists as a field in the Firestore document being decoded.")
+        let docPath = (docRef != nil) ? docRef!.path : "being read"
```

Comment:
"being read" doesn't make any sense to me--though I'm probably misunderstanding your intent.

What about "nil" or "none"?

Reply:
Done.